clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java

```diff
@@ -522,6 +528,39 @@ private void checkHeartbeatTimeout() {
     }
   }
 
+  void reportMetricsUpdateToMainProcess() {
+    try {
+      IngestionMetricsReport metricsReport = new IngestionMetricsReport();
+      metricsReport.aggregatedMetrics = new VeniceConcurrentHashMap<>();
+      if (getMetricsRepository() != null) {
+        getMetricsRepository().metrics().forEach((name, metric) -> {
+          if (metric != null) {
+            try {
+              // Best-effort to reduce metrics delta size sent from child process to main process.
+              Double originalValue = getMetricsMap().get(name);
```

Comment:
This has the assumption that the metric reporting request will always be successful, right?
And if it fails, the next metric report request could miss some metrics included in the errored request.

Reply:
This is fine at this moment since the server won't stop processing even client timeout happens.
Let us add some TODO here to indicate there is a risk.

Reply:
Sure.